File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/search/MultipleQueriesStrategy.java
package com.algolia.model.search;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** Gets or Sets multipleQueriesStrategy */
@JsonAdapter(MultipleQueriesStrategy.Adapter.class)
public enum MultipleQueriesStrategy {
  NONE("none"),

  STOPIFENOUGHMATCHES("stopIfEnoughMatches");

  private String value;

  MultipleQueriesStrategy(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static MultipleQueriesStrategy fromValue(String value) {
    for (MultipleQueriesStrategy b : MultipleQueriesStrategy.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<MultipleQueriesStrategy> {

    @Override
    public void write(
      final JsonWriter jsonWriter,
      final MultipleQueriesStrategy enumeration
    ) throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public MultipleQueriesStrategy read(final JsonReader jsonReader)
      throws IOException {
      String value = jsonReader.nextString();
      return MultipleQueriesStrategy.fromValue(value);
    }
  }
}

File: clients/algoliasearch-client-php/lib/Model/Search/MultipleQueriesStrategy.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * MultipleQueriesStrategy Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class MultipleQueriesStrategy
{
    /**
     * Possible values of this enum
     */
    const NONE = 'none';

    const STOP_IF_ENOUGH_MATCHES = 'stopIfEnoughMatches';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::NONE,
            self::STOP_IF_ENOUGH_MATCHES,
        ];
    }
}

